clion/src/main/java/org/sonarlint/intellij/clion/CFamilyAnalysisConfigurator.java

```diff
@@ -47,4 +52,13 @@ public AnalysisConfiguration configure(Module module, Collection<VirtualFile> fi
     return result;
   }
 
+  private static boolean isSupported() {
+    try {
+      Class.forName("com.jetbrains.rider.cpp.fileType.psi.CppFile");
+      return false;
+    } catch (ClassNotFoundException e) {
+      return true;
+    }
+  }
```

Comment:
Hmm I find this a bit surprising: if the class is present, then we consider it is not supported? Are the return values inverted, or should the method name be improved?

Reply:
CppFile is only present when ReSharper is on, so when this is the case I don't want to run the regular CLion analysis

Reply:
Then maybe we could find a more informative name for the method, because `isSupported` does not tell what is supported. Something like `isClionResharperOn()`? And then invert the logic